golang1.17/bin/compile

```diff
@@ -86,32 +86,44 @@ def build(source_dir, target_dir):
     }
 
     gomod = "%s/go.mod" % source_dir
+    vendor = "%s/vendor" % source_dir
+
+    has_gomod = exists(gomod)
+    deps_vendored = exists(vendor)
+
+    if has_gomod:
+        print("Detected go.mod file.")
+    if deps_vendored:
+        print("Detected vendor directory. Will skip downloading modules.")
+
     if os.environ.get("__NIM_REMOTE_BUILD"):
-        if exists(gomod):
-            print("downloading modules")
-            ret = subprocess.call(["go", "mod", "download"], cwd=source_dir, env=env)
-            if ret != 0:
-                sys.exit(ret)
+        if has_gomod:
+            if not deps_vendored:
+                print("downloading modules")
+                ret = subprocess.call(["go", "mod", "download"], cwd=source_dir, env=env)
+                if ret != 0:
+                    sys.exit(ret)
```

Comment:
Do we need this? Doesn't the `go build` command below download the dependencies anyway?

Reply:
It does. I did it this way to change the way it worked as little as possible. I decided it was still okay if it downloads modules early, as long as it skipped that step if a vendor directory was present.

Reply:
wfm